satpy/readers/seviri_l1b_hrit.py

```diff
@@ -626,6 +629,11 @@ def get_dataset(self, key, info):
         """Get the dataset."""
         res = super(HRITMSGFileHandler, self).get_dataset(key, info)
         res = self.calibrate(res, key['calibration'])
+        if (key['calibration'] in ['radiance', 'reflectance', 'brightness_temperature'] and
+            self.mask_bad_quality_scan_lines):  # noqa: E129
```

Comment:
I totally agree that this is not really nice and readability is somewhat impaired. I have a feeling that this change request is also inspired by the "Clean Code" book ;-). In this case I have to strongly disagree though that refactoring this into a separate method would improve readability. This would fragment the code a lot with yet another private method which has no real purpose. While it might improve the one line ,the "unattractive" code would just be moved somewhere else and a reader would need to jump to a method with no body just to have a look at what conditions are applied here. My proposition (which is included in the last push) is to move the list check to a separate line which improves readability and keeps the code lean an close together.

Reply:
I disagree (maybe I've watched too many Uncle Bob videos), but your proposed solution sounds fine! :)